modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryFutureAdapter.java

```diff
@@ -480,12 +480,12 @@ void clear() {
     /** {@inheritDoc} */
     @Override public void onTimeout() {
         try {
-            cancelQuery();
-
             onDone(new IgniteFutureTimeoutCheckedException("Query timed out."));
+
+            cancelQuery();
```

Comment:
Is there any reason to change the order?
However, both ways look correct.

Reply:
It was discussion about it in the core chat. @ygerzhedovich advised to do so. 

Inside cancelQuery there is `assert isDone() : this;` I And in my test I was getting AssertionError

Reply:
Thanks, got it.